train_models_given_z.py

```diff
@@ -0,0 +1,254 @@
+"""
+2018 Gregory Way
+train_models_given_z.py
+
+This script will loop through different combinations of latent space
+dimensionality, train a distinct model, and save associated decoder weights
+and z matrices. The script will need to pull appropriate hyperparameters from
+respective files after several initial sweeps testing various hyperparameter
+combinations with different z dimensionality.
+
+The script will fit several different compression algorithms and output
+results. The results include the weight matrices and z matrices of the models
+with the lowest reconstruction loss. The total stability of reconstruction
+losses are represented through the variability associated with each model after
+retraining several times given the argument for number of seeds. We also
+report the determinants of correlation matrices for all iterations. These
+values measure the stability of either latent space (z) matrices or weight
+matrices across several rounds of fitting. In this case, as the determinants
+approach zero, the more stable solutions are.
+
+Usage:
+
+    python run_model_with_input_dimensions.py
+
+    With required command line arguments:
+
+        --num_components    The z dimensionality we're testing
+        --param_config      A tsv file (param by z dimension) indicating the
+                            specific parameter combination for the z dimension
+        --out_dir           The directory to store the output files
+
+    And optional command line arguments
+
+        --num_seeds         The number of specific models to generate
+                                default: 5
+"""
+
+import os
+import argparse
+import numpy as np
+import pandas as pd
+from tybalt.data_models import DataModel
+
+
+def get_lowest_loss(matrix_list, reconstruction_df,
+                    algorithms=['pca', 'ica', 'nmf', 'dae', 'vae']):
+    """
+    Determine the specific model with the lowest loss using reconstruciton cost
+
+    Arguments:
+    matrix_list - list of matrices (either weight or z matrices)
+    reconstruction_df - pandas DataFrame (seed by reconstruction error)
+    algorithms - which algorithms to consider
+
+    Output:
+    Single matrix of the "best" alternative matrices by lowest recon error
+    """
+    final_matrix_list = []
+    for alg in algorithms:
+        # Get the lost reconstruction cost for given algorithm
+        min_recon_subset = reconstruction_df.loc[:, alg].idxmin()
```

Comment:
Just checking my understanding here -- you'll be looking for the minimum loss value of the _final_ loss values at end of training, is that correct?

Reply:
I actually pass the data through the bottleneck (of each algorithm) and reconstruct. Then I take the difference between the input and the output.

Reply:
i agree the comment makes this confusing... Maybe "Get the lowest reconstruction cost _across iterations_ for the given algorithm"